pallets/capacity/src/lib.rs

```diff
@@ -913,6 +914,14 @@ impl<T: Config> Pallet<T> {
 
 		capacity_details.withdraw(actual_capacity, actual_amount);
 
+		if staking_type.eq(&ProviderBoost) {
+			// update staking history
+			let era = Self::get_current_era().era_index;
+			let mut pbh =
+				Self::get_staking_history_for(unstaker).ok_or(Error::<T>::NotAStakingAccount)?;
+			pbh.subtract_era_balance(&era, &actual_amount);
```

Comment:
nit: this check can move above get_current_era , not sure if that function makes a read so we might save one one read